fastlane/Fastfile

```diff
@@ -58,37 +72,114 @@ lane :release do |options|
   #pod_lib_lint(podspec: "StreamChat.podspec", allow_warnings: true)
   #pod_lib_lint(podspec: "StreamChatUI.podspec", allow_warnings: true)
 
+  # Bump Podspec Version
   version_bump_podspec(path: "StreamChat.podspec", version_number: version_number)
   version_bump_podspec(path: "StreamChatUI.podspec", version_number: version_number)
 
+  # Bump XCFramework Version
+  version_bump_podspec(path: "StreamChat-XCFramework.podspec", version_number: version_number)
+  version_bump_podspec(path: "StreamChatUI-XCFramework.podspec", version_number: version_number)
+
+  # Stage all changes
   sh("git add -A")
 
   if(!prompt(text: "Will commit changes. All looking good?", boolean: true))
     UI.user_error!("Not committing changes")
   end
 
   sh("git commit -m 'Bump #{version_number}'")
-  sh("git tag #{version_number}")
 
   if(!prompt(text: "Will push changes. All looking good?", boolean: true))
     UI.user_error!("Not pushing changes")
   end
 
-  push_to_git_remote(tags: true)
+  push_to_git_remote(tags: false)
+
+  # Open a new PR
+  create_pull_request(
+    api_token: ENV["GITHUB_TOKEN"],
+    repo: "GetStream/stream-chat-swift",
+    title: "#{version_number} Release",
+    head: "release/#{version_number}",
+    base: "main",
+    body: "#{changes}"
+)
+
+  UI.success("Successfully started release #{version_number}")
+end
+
+desc "Completes an SDK Release"
+lane :publish_release do |options|
 
+  version = options[:version]
+
+  # Make xcframeworks
+  sh("cd .. && make frameworks")
+
+  # Generate Checksums
+  streamChatChecksum = sh("swift package compute-checksum ../Products/StreamChat.zip").strip
+  streamChatUIChecksum = sh("swift package compute-checksum ../Products/StreamChatUI.zip").strip
+
+  # Update SPM Repo
+  sh("git clone git@github.com:GetStream/stream-chat-swift-spm.git ../StreamSPM")
+  file = File.open("../StreamSPM/Package.swift")
+  file_data = file.read
+
+  # [WIP] We can improve using regex here
+  fileLines = File.readlines("../StreamSPM/Package.swift")
+
+  # We are unfortunately having to dig into a file and manually change the lines required
+  fileLines.each_with_index do |line, index|
+  if line.include?("url") && line.include?("StreamChat.zip")
+      streamChat = "url: \"https://github.com/GetStream/stream-chat-swift/releases/download/#{version}/StreamChat.zip\","
+      file_data.gsub!(line, streamChat)
+
+      checksumItem = fileLines[index + 1]
+      newChecksumLine = "checksum: \"#{streamChatChecksum}\"),"
+      file_data.gsub!(checksumItem, newChecksumLine)
+  end
+  if line.include?("url") && line.include?("StreamChatUI.zip")
+      streamChatUI = "url: \"https://github.com/GetStream/stream-chat-swift/releases/download/#{version}/StreamChatUI.zip\","
+      file_data.gsub!(line, streamChatUI)
+
+      newUIChecksumItem = fileLines[index + 1]
+      newUIChecksumLine = "checksum: \"#{streamChatUIChecksum}\"),"
+      file_data.gsub!(newUIChecksumItem, newUIChecksumLine)
+  end
+end
```

Comment:
this `end` should be removed

Reply:
It's required for the end of the loop 🤔

Reply:
oh, sorry! the indentation is incorrect, so I missed the loop

Reply:
Yeah, the good point though on indentation... let me fix that one!

Reply:
There is a problem on the indentation starting line 133